components/nimbus/src/tests/stateful/test_nimbus.rs

```diff
@@ -1686,3 +1687,59 @@ fn test_recorded_context_recorded() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_recorded_context_event_queries() -> Result<()> {
+    let metrics = TestMetrics::new();
+
+    let temp_dir = tempfile::tempdir()?;
+
+    let app_context = AppContext {
+        app_name: "fenix".to_string(),
+        app_id: "org.mozilla.fenix".to_string(),
+        channel: "nightly".to_string(),
+        app_version: Some("124.0.0".to_string()),
+        ..Default::default()
+    };
+    let recorded_context = Arc::new(TestRecordedContext::new());
+    recorded_context.set_context(json!({
+        "app_version": "125.0.0",
+        "other": "stuff",
+    }));
+    recorded_context.set_event_queries(Map::from_iter(vec![(
+        "TEST_QUERY".to_string(),
+        serde_json::to_value("'event'|eventSum('Days', 1, 0)")?,
+    )]));
+    let client = NimbusClient::new(
+        app_context.clone(),
```

Comment:
You don't need to cloen this. It can take ownership since we're not using app_context below here.